x-pack/plugins/apm/public/components/shared/charts/timeseries_chart.tsx

```diff
@@ -86,16 +86,47 @@ export function TimeseriesChart({
 
   const xFormatter = niceTimeFormatter([min, max]);
 
-  const isEmpty = timeseries
-    .map((serie) => serie.data)
-    .flat()
-    .every(
-      ({ y }: { x?: number | null; y?: number | null }) =>
-        y === null || y === undefined
-    );
+  const data = Array.isArray(timeseries)
+    ? timeseries?.map((serie) => serie.data).flat()
+    : timeseries?.data;
+
+  const isEmpty = data
+    ? data.every(
+        ({ y }: { x?: number | null; y?: number | null }) =>
+          y === null || y === undefined
+      )
+    : true;
```

Comment:
Is this something that we can outsource to Elastic charts?

Reply:
It's used by the chart container to determine whether to show the loading placeholder: https://github.com/elastic/kibana/blob/9527e0671dbe70e2e4cdcf27d6d7c592f6ec57d0/x-pack/plugins/apm/public/components/shared/charts/chart_container.tsx#L20-L25